app/main/lib/elasticsearch.py

```diff
@@ -45,19 +45,24 @@ def get_all_documents_matching_context(context):
     return []
 
 def generate_matches(context):
+    """
+        If the keys are not project_media_id, has_custom_id, or field, return ANDs for each field,
+        with ORs for intra-key values (e.g. foo = bar AND baz = (blah|bat))
```

Comment:
Aside: does this mean if I'm trying to restrict the search space by adding additional facets in the context, it will OR the result set?

Reply:
@skyemeedan it will OR values in a specific key, and AND across keys. So if you were trying to filter baz by blah *and* bat, it would not interpret that correctly and instead to blah|bat. Does that help at all?

Reply:
yeah I understand that, but just not clear what the reason for it.  It seems like, if I was trying to reduce the size of the result set to process (to avoid OpenSearch heap errors for example)  adding additional keys would expand the search space instead of reducing it.  There must be a feature that needs it to work this way?